contract/models/res_partner.py

```diff
@@ -46,14 +46,15 @@ def act_show_contract(self):
         contract_type = self._context.get('contract_type')
 
         res = self._get_act_window_contract_xml(contract_type)
-        res.update(
-            context=dict(
-                self.env.context,
-                search_default_partner_id=self.id,
-                default_partner_id=self.id,
-                default_pricelist_id=self.property_product_pricelist.id,
-            ),
-        )
+        t_context = dict(
+            self.env.context,
+            search_default_partner_id=self.id,
+            default_partner_id=self.id,
+            default_pricelist_id=self.property_product_pricelist.id,
+            )
+        if 'group_by' in t_context:
+            del t_context['group_by']
+        res.update(context=t_context)        
```

Comment:
You can gain a few lines:

```suggestion
action_context = {k: v for k, v in self.env.context if k != "groupby"}
action_context["search_default_partner_id"] = self.id
action_context["default_partner_id"] = self.id
action_context["default_pricelist_id"] = self.property_product_pricelist.id

res["context"] = action_context  
```
(or you could simply use `t_context.pop('group_by', None)`).
Anyway, LGTM.